src/glfw3.jl

```diff
@@ -289,12 +290,31 @@ immutable VidMode
 	refreshrate::Cint   # The refresh rate, in Hz, of the video mode.
 end
 
+immutable GLFWError <: Exception
+	code::Cint
+	description::String
+end
+Base.showerror(io::IO, e::GLFWError) = print(io, "GLFWError: ", e.description)
+
 #************************************************************************
 # GLFW API functions
 #************************************************************************
 
 # Initialization and version information
-Init() = Bool(ccall( (:glfwInit, lib), Cint, ())) || error("initialization failed")
+function Init()
+	try
+		return Bool(ccall( (:glfwInit, lib), Cint, ())) || error("initialization failed")
+	catch ex
+		if isa(ex, GLFWError) && ex.code == PLATFORM_ERROR && contains(ex.description, "Failed to get display service port iterator")
```

Comment:
Ha, almost forgot, there is still this one: https://github.com/glfw/glfw/blob/798d7c6d68bb17480fcda6074b1d5a946a05ed52/src/linux_joystick.c#L143
I guess you run into that one on Travis as well? Jeez, it has been way too long that I lived with the assumption, that these errors are fatal :D